### PR TITLE
Add binding to biginteger factorial in GMP

### PR DESCRIPTION
**Pull request by peheje**
Add binding to biginteger factorial in GMP

### CODE REVIEW

**Comment by peheje** [comment]
https://github.com/crystal-lang/crystal/issues/9072

**Comment by peheje** [comment]
Thanks for review RX14.
Added the test `elsif self > LibGMP::ULong::MAX`

If acceptable, please merge as I do not have authorization. Else let me know what to change.

**Comment by RX14** [comment]
Looks like this needs a format actually @peheje, please could you run `bin/crystal tool format` inside the crystal repo and commit the changed.

**Comment by peheje** [comment]
I ran the format, thanks for this, nice tool :-)

File: spec/std/big/big_int_spec.cr [hunk 1]
```diff
@@ -145,6 +145,28 @@ describe "BigInt" do
     (-10.to_big_i.abs).should eq(10.to_big_i)
   end
 
+  it "gets factorial value" do
+    (0.to_big_i.factorial).should eq(1.to_big_i)
+    (5.to_big_i.factorial).should eq(120.to_big_i)
+    (100.to_big_i.factorial).should eq("93326215443944152681699238856266700490715968264381621468592963895217599993229915608941463976156518286253697920827223758251185210916864000000000000000000000000".to_big_i)
```
Comment (Sija) [inline spec/std/big/big_int_spec.cr:151]:
```suggestion
    0.to_big_i.factorial.should eq(1.to_big_i)
    5.to_big_i.factorial.should eq(120.to_big_i)
    100.to_big_i.factorial.should eq("93326215443944152681699238856266700490715968264381621468592963895217599993229915608941463976156518286253697920827223758251185210916864000000000000000000000000".to_big_i)
```

File: spec/std/big/big_int_spec.cr [hunk 2]
```diff
@@ -145,6 +145,28 @@ describe "BigInt" do
...
+  end
+
+  it "raises if factorial of negative" do
+    expect_raises ArgumentError do
+      (-1.to_big_i.factorial)
```
Comment (Sija) [inline spec/std/big/big_int_spec.cr:156]:
```suggestion
      -1.to_big_i.factorial
```

File: spec/std/big/big_int_spec.cr [hunk 3]
```diff
@@ -145,6 +145,28 @@ describe "BigInt" do
...
+    end
+
+    expect_raises ArgumentError do
+      ("-93326215443944152681699238856266700490715968264381621468592963895217599993229915608941463976156518286253697920827223758251185210916864000000000000000000000000".to_big_i.factorial)
```
Comment (Sija) [inline spec/std/big/big_int_spec.cr:160]:
```suggestion
      "-93326215443944152681699238856266700490715968264381621468592963895217599993229915608941463976156518286253697920827223758251185210916864000000000000000000000000".to_big_i.factorial
```

File: spec/std/big/big_int_spec.cr [hunk 4]
```diff
@@ -145,6 +145,28 @@ describe "BigInt" do
...
+    end
+  end
+
+  it "raises if factorial of 2^64" do
+    expect_raises ArgumentError do
+      ((((LibGMP::ULong::MAX).to_big_i) + 1).factorial)
```
Comment (Sija) [inline spec/std/big/big_int_spec.cr:166]:
```suggestion
      (LibGMP::ULong::MAX.to_big_i + 1).factorial
```

**Pull request merged**